### ISSUE TITLE
Optical circuits and Advanced AssLine

### ISSUE THREAD

**Issue by RaitGamerGR** [GTNewHorizons/GT-New-Horizons-Modpack#13207]
### Your GTNH Discord Username

Rait_GamerGR(Александр)#5721

### Your Pack Version

2.3.2

### Your Server

Private Server

### Java Version

Java 8

### Type of Server

Ultramine

### Your Expectation

Trying to craft Optical Computer in Advanced AssLine

### The Reality

First item and fluids consumed, but machine stucks on first slice

![2023-04-13_10 07 14](https://user-images.githubusercontent.com/70205464/231682030-c89ba9b6-c12a-4423-b75b-f12eefa94e1d.png)

As i can see, Advanced AssLine check only first item and fluid to start recipe, so it starts Optical Computer craft (same first item and sluids, but with half amount.

![image](https://user-images.githubusercontent.com/70205464/231682984-fe9a21ff-e72d-4dff-a065-34653e2cf106.png)
![image](https://user-images.githubusercontent.com/70205464/231683030-b86feb98-3e99-4a05-a95f-169888b1afa3.png)

### Your Proposal

Change Optical Ciruits crafts or change Advances AssLine recipe check function 

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [X] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by Glease** [comment]
maybe the first slice should only start if all other slices can start?

**Comment by Glease** [comment]
So I get back and got a bit more thinking to this. This is actually a known issue that I'm very reluctant at solving. For example, you put in 10 optical computer worth of material and 10 optical assembly worth of material simultaneously. Whichever recipe got executed first (let's assume optical computer for the sake of argument), when the first slice finished 10 craft, it looked at its input and still see matching input. Then, it **continues to consume circuit board and keeps on doing optical computer recipe**, even if in an ideal world it should stop right there OR switch to the other recipe. 

This is an additional automation constraint I added, so you either 
* use one recipe per AAL
* use only nonconflicting recipes per AAL and ensure you queued the stuff to input bus properly
* invent smart control that pause putting stuff into input bus until the last recipe type is fully finished.

I'd very much like to see people doing item 3 (and to that end I made AAL extremely cheap), but so far people have been unable/unwilling to come up with a solution.

Yes, I can remove this artificial constraint by allowing to switch to other recipe to try to resolve a STUCK AAL, likely by adding a upgrade chip, but I'll make that upgrade chip extremely expensive and the AAL controller itself moderately more expensive. This however trivialize AAL into just another mega machine - high price high throughput and nothing more. It helps you make your next stargate, but it's just more endgame power creep instead of actual meaningful content.

The deal here is simple: choose one of these three
* suffer the penalty of one recipe/nonconflict recipe per AAL
* get smart and play the logistics puzzle
* stuff get expensive 

I haven't yet decided on how expensive it should be, but the goal is to make people that use this path regret it.

**Comment by RaitGamerGR** [comment]
I invented automation to use AAL like normal AL for any recipes. This is only two conflicted recipes i found. There is no automation way to avoid stucking of this recipe, as i think, because my automation put new one recipe only if AAL have no fluids->recipe started. I think now i have only 1 way to avoid it, remove one of data sticks with conflicted recipes.

**Comment by Ydrec** [comment]
from my experience AAL needs a whole recipe inside to start, i haven't seen it be able to start with just the first item/fluid inside. Maybe when its already started it only checks specific slice item

**Comment by Glease** [comment]
Yeah Ydrec is correct. It doesn't *start*, but it can *continue*.

**Comment by Ydrec** [comment]
Yeah seems like the issue is AAS doesn't do a proper recipe check when first slice is depleted instead just continuing to do same recipe if same item is present like other slices.

This would happen to any recipe in which fluids/first item is same.

While its possible to invent smart control it would need to either:
- Not put next recipe until all slices are depleted, basically removing parallels.
- Remember every single used recipe to compare inputs and not push different recipe, which is pain to setup on one, less multiple lines.
